test/objective.jl

```diff
@@ -64,6 +64,20 @@ function objectives_test(ModelType::Type{<:JuMP.AbstractModel}, VariableRefType:
             4x + 1, @inferred JuMP.objective_function(m, AffExprType))
     end
 
+    @testset "Linear objective changes" begin
+        m = ModelType()
+        @variable(m, x)
+
+        # No change possible for a single variable.
+        @objective(m, Max, x)
+        @test_throws MethodError set_objective_coefficient(m, x, 4.0)
```

Comment:
Hmm. Users could reasonably expect that this would work. I would suggest that the solvers promote the objective to `ScalarAffine`.

Reply:
I did that at the JuMP level -- unless there is a reason to let the solvers handle that case?